### ISSUE TITLE
add option to disable the upgrade_check by default

### ISSUE THREAD

**Issue by 0-wiz-0** [JanDeDobbeleer/oh-my-posh#5387]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What would you like to see added?

Thank you for oh-my-posh!
I'm a packager for pkgsrc, and by default we disable all "phone home" features, as benevolent as they might be.
I noticed that oh-my-posh has an upgrade notice that is enabled by default.
I found that I can disable it for a user by "oh-my-posh disable notice", and I found the code in `src/upgrade/notice.go`, so I can disable the functionality in the package completely by adding a `return "", false` early in that function.
But I didn't find a good way to disable it by default (so that users can still enable it with "oh-my-posh enable notice" if they want it).
Can you please add support for packaging it with the notice disabled by default?

**Comment by JanDeDobbeleer** [comment]
@0-wiz-0 I could add a compiler flag in case you're compiling this yourself, would that work?

**Comment by 0-wiz-0** [comment]
Sure, that'd be great!

**Comment by 0323pin** [comment]
@JanDeDobbeleer 

NetBSD package maintainer here. I see that this has been addressed in https://github.com/JanDeDobbeleer/oh-my-posh/commit/9e238adbb9db5079cbf4119531709591a626af2e
But, is the the notice feature disabled by default now? What's the compiler flag to be used to enable/disable the feature? Should I pass the equivalent to `go:build disablenotice` to the compiler?

Thanks!

**Comment by JanDeDobbeleer** [comment]
@0323pin no, still enabled by default unless you compile with `--tags=disablenotice`

**Comment by 0323pin** [comment]
Thank you!

**Comment by 0-wiz-0** [comment]
@JanDeDobbeleer Thanks for the patch.

I'm a bit confused. I tried passing tags to the build to get it to compile the new version, but after every restart of oh-my-posh (after removing `.cache/oh-my-posh`), my `omp.cache` looked like this:
```
{
    "is_wsl": {
        "value": "false",
        "timestamp": 1723013696,
        "ttl": -1
    },
    "upgrade_check": {
        "value": "v23.6.0",
        "timestamp": 1723013698,
        "ttl": 10080
    }
}
```
which *looks* like it did check for a new version.

I don't know much about tags, so I thought I had made an error there, so I just made both implementations the same:
```
--- src/config/notice_default.go 2024-08-07 08:54:14.549759028 +0200
+++ src/config/notice_disable.go 2024-08-06 18:22:45.000000000 +0200
@@ -1,4 +1,4 @@
-//go:build !disablenotice
+//go:build disablenotice

 package config
```
to get the new behaviour independent of my inability to correctly pass tags.
But even after installing and starting such a version, I see the same `omp.cache` file.

To compare, when I run `oh-my-posh disable notice`, the file looks like this:
```
{
    "is_wsl": {
        "value": "false",
        "timestamp": 1723013696,
        "ttl": -1
    },
    "upgrade_check": {
        "value": "disabled",
        "timestamp": 1723013891,
        "ttl": -1
    }
}
```
which is what I would have expected the new version to generate too.

Am I misunderstanding something or is there a bug?
Thanks.

**Comment by JanDeDobbeleer** [comment]
I'll have a look.

**Comment by 0323pin** [comment]
@JanDeDobbeleer As this issue is closed, you might have simply forgotten about it. Or, you might just be on holiday like a big chunk of Europe right now.

If the later, sorry for bumping.

**Comment by JanDeDobbeleer** [comment]
It's still in my to-do list, but yes, holidays

**Comment by JanDeDobbeleer** [comment]
@0-wiz-0 when I use this flag, and there's no `auto_upgrade: true`, nor `disable_notice: false` override, this works perfectly for me. That code is never added to the init scripts and thus never executed.

**Comment by 0-wiz-0** [comment]
@JanDeDobbeleer 
I'm confused about how the phone-home is disabled, and I wasn't sure how to correctly add `disablenotice` in our builds. So I made a local change to make `disablenotice` the default:
```
--- config/notice_default.go.orig       2024-08-09 09:37:05.000000000 +0000
+++ config/notice_default.go
@@ -2,4 +2,14 @@

 package config

-func setDisableNotice(_ []byte, _ *Config) {}
+import "bytes"
+
+func setDisableNotice(stream []byte, cfg *Config) {
+       if !hasKeyInByteStream(stream, "disable_notice") {
+               cfg.DisableNotice = true
+       }
+}
+
+func hasKeyInByteStream(data []byte, key string) bool {
+       return bytes.Contains(data, []byte(key))
+}
```
I.e. the default `!disablenotice` (NO disablenotice) in that version was the same code as the `disablenotice` (YES disablenotice) one.
I removed `$HOME/.cache/oh-my-posh` and traced the syscalls of a newly started zsh where I manually ran `eval "$(oh-my-posh init zsh)"`.
The syscall output contained IP addresses of github.com. Why did they appear there? I had hoped that there would be no need for those.

Anyway, I've stopped using oh-my-posh in the meantime, so I'll not pursue this from my side, but let me know if you have any questions.
